satpy/writers/cf_writer.py

```diff
@@ -482,18 +500,30 @@ def update_encoding(dataset, to_netcdf_kwargs):
     other_to_netcdf_kwargs = to_netcdf_kwargs.copy()
     encoding = other_to_netcdf_kwargs.pop('encoding', {}).copy()
 
+    _set_encoding_dataset_names(encoding, dataset, numeric_name_prefix)
     _set_default_chunks(encoding, dataset)
     _set_default_fill_value(encoding, dataset)
     _set_default_time_encoding(encoding, dataset)
 
     return encoding, other_to_netcdf_kwargs
 
 
+def _handle_dataarray_name(original_name, numeric_name_prefix):
+    name = original_name
+    if name[0].isdigit():
+        if numeric_name_prefix:
+            name = numeric_name_prefix + original_name
+        else:
+            warnings.warn('Invalid NetCDF dataset name: {} starts with a digit.'.format(name))
+    return original_name, name
+
+
 class CFWriter(Writer):
     """Writer producing NetCDF/CF compatible datasets."""
 
     @staticmethod
-    def da2cf(dataarray, epoch=EPOCH, flatten_attrs=False, exclude_attrs=None, compression=None):
+    def da2cf(dataarray, epoch=EPOCH, flatten_attrs=False, exclude_attrs=None, compression=None,
+              include_orig_name=False, numeric_name_prefix='CHANNEL_'):
```

Comment:
I think `include_orig_name` should be True by default. Or do you foresee problems with that?

Reply:
No I don't see any problems with that. It's just how I would prefer it. I will need to update some of the tests I think when I change this.

Reply:
Ah, it doesn't really matter, do as you prefer.